modules/core/src/test/scala/com/azavea/stac4s/CatalogLayerSpec.scala

```diff
@@ -0,0 +1,438 @@
+package com.azavea.stac4s
+
+import com.azavea.stac4s.extensions.layer._
+import io.circe.syntax._
+import cats.syntax.either._
+import geotrellis.vector._
+
+import java.time._
+
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+
+class CatalogLayerSpec extends AnyFunSpec with Matchers {
+  import JsonUtils._
+
+  describe("CatalogLayerSpec") {
+    it("Create LC8 Layers Catalog") {
+      val root =
+        StacCatalog(
+          id = "landsat-stac-layers",
+          stacVersion = "0.9.0",
+          stacExtensions = Nil,
+          title = Some("STAC for Landsat data"),
+          description = "STAC for Landsat data",
+          links = List(
+            StacLink(
+              href = "./catalog.json",
+              rel = StacLinkType.Self,
+              _type = None,
+              title = None,
+              // should it be an optional thing?
+              labelExtAssets = Nil
+            ),
+            StacLink(
+              href = "./catalog.json",
+              rel = StacLinkType.StacRoot,
+              _type = None,
+              title = None,
+              // should it be an optional thing?
+              labelExtAssets = Nil
+            ),
+            StacLink(
+              href = "./landsat-8-l1/catalog.json",
+              rel = StacLinkType.Child,
+              _type = None,
+              title = None,
+              // should it be an optional thing?
+              labelExtAssets = Nil
+            ),
+            StacLink(
+              href = "./layers/ca/catalog.json",
+              rel = StacLinkType.Child,
+              _type = None,
+              title = None,
+              // should it be an optional thing?
+              labelExtAssets = Nil
+            ),
+            StacLink(
+              href = "./layers/us/catalog.json",
+              rel = StacLinkType.Child,
+              _type = None,
+              title = None,
+              // should it be an optional thing?
+              labelExtAssets = Nil
+            )
+          )
+        )
+
+      root.asJson.deepDropNullValues shouldBe getJson("/catalogs/landsat-stac-layers/catalog.json")
+      root.asJson.as[StacCatalog].valueOr(throw _) shouldBe root
+    }
+
+    it("Create LC8 Layer Collection") {
+      val layerUS = StacCollection(
```

Comment:
I guess this is incorrect, layers can't be collections, since items should have a back reference to al the collections in a collection field. 

I'll make the catalogs and we'll see what would happen.